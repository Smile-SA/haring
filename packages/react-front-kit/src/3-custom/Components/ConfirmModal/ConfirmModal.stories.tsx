import type { Meta, StoryObj } from '@storybook/react';

import { MANTINE_COLORS } from '@mantine/styles';

import { ConfirmModal as Cmp } from './ConfirmModal';

const colorOptions = ['primary'].concat(MANTINE_COLORS);

const meta = {
  argTypes: {
    cancelColor: {
      control: 'select',
      options: colorOptions,
    },
    confirmColor: {
      control: 'select',
      options: colorOptions,
    },
    onClick: { action: 'clicked' },
    opened: { control: 'boolean' },
  },
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/ConfirmModal',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const ConfirmModal: IStory = {
  args: {
    cancelColor: 'gray',
    cancelLabel: 'Cancel',
    children: 'Are you sure you want to delete this item?',
    confirmColor: 'red',
    confirmLabel: 'Remove',
    opened: false,
    title: 'Remove ?',
  },
};
