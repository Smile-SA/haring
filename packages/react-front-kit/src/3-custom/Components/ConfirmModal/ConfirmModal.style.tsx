import { createStyles } from '@mantine/styles';

export const useStyles = createStyles(() => ({
  buttonFilters: { background: 'white', height: '34px', marginRight: '10px' },
  buttonGrey: {
    padding: '0.667em 3.333em',
  },
  buttonLeftModal: { marginBottom: '10px', marginRight: '10px' },
  buttonRemoveRoot: {
    padding: '0.667em 3.333em',
  },
  modalBody: {
    padding: '0px',
  },
  modalButtonsContainer: {
    marginTop: '32px',
  },
  modalContent: {
    padding: '48px',
  },
  modalHeader: {
    height: '0px',
    padding: '0px',
  },
  modalTitleContainer: {
    marginLeft: '12px',
  },
}));
