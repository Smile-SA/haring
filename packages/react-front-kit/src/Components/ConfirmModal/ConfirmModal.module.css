.buttonLeftRoot {
  padding: 0.667em 3.333em;
}

.buttonLeftModal {
  margin-bottom: 10px;
  margin-right: 10px;
}

.buttonRightRoot {
  padding: 0.667em 3.333em;
}

.modalBody {
  padding: 0;
}

.modalButtonsContainer {
  margin-top: 32px;
}

.modalContent {
  padding: 48px;
}

.modalHeader {
  height: 0;
  padding: 0;
}

.modalTitleContainer {
  margin-left: 12px;
}

.title {
  font-size: 26px;
  font-weight: 700;
}

.cross {
  position: absolute;
  top: -25px;
  right: -25px;
  transform: scale(115%);
}
