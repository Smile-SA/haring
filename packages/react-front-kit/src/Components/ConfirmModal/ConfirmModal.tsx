'use client';

import type { IConfirmModal } from '@smile/haring-react-shared';
import type { ReactElement } from 'react';

import { Button, Flex, Modal } from '@mantine/core';

import classes from './ConfirmModal.module.css';

export type IConfirmModalProps = IConfirmModal;

export function ConfirmModal(props: IConfirmModalProps): ReactElement {
  const {
    cancelColor = 'gray',
    cancelLabel = 'Cancel',
    confirmColor = 'primary',
    confirmLabel = 'Confirm',
    onCancel,
    onClose,
    onConfirm,
    children,
    title,
    ...modalProps
  } = props;

  return (
    <Modal
      centered
      classNames={{
        body: classes.modalBody,
        close: classes.cross,
        content: classes.modalContent,
        header: classes.modalHeader,
        title: classes.title,
      }}
      onClose={onClose}
      radius={16}
      size="lg"
      title={title}
      {...modalProps}
    >
      <>
        {children}
        <Flex className={classes.modalButtonsContainer} wrap="wrap">
          <Button
            className={classes.buttonLeftModal}
            classNames={{
              root: classes.buttonLeftRoot,
            }}
            color={cancelColor}
            onClick={onCancel}
          >
            {cancelLabel}
          </Button>
          <Button
            classNames={{
              root: classes.buttonRightRoot,
            }}
            color={confirmColor}
            onClick={onConfirm}
          >
            {confirmLabel}
          </Button>
        </Flex>
      </>
    </Modal>
  );
}
