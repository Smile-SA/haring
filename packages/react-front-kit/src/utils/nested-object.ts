export type INestedObjectWithDepth<O extends object> = O & {
  children?: INestedObjectWithDepth<O>[];
  depth?: number;
};

export function flattenNestedObjects<O extends object>(
  arr: INestedObjectWithDepth<O>[],
): INestedObjectWithDepth<O>[] {
  let children: INestedObjectWithDepth<O>[] = [];
  const flattened = arr.map((m) => {
    if (m.children && m.children.length) {
      children = [...children, ...m.children];
    }
    return m;
  });
  return flattened.concat(
    children.length ? flattenNestedObjects(children) : children,
  );
}

export function assignDepth<O extends object>(
  arr?: INestedObjectWithDepth<O>[],
  depth = 0,
): INestedObjectWithDepth<O>[] {
  (arr ?? []).forEach((o) => {
    o.depth = depth;
    assignDepth(o.children, depth + 1);
  });
  return arr ?? [];
}
