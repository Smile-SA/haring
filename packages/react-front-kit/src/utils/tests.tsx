import type {
  Queries,
  RenderOptions,
  RenderResult,
  queries,
} from '@testing-library/react';
import type { ReactElement, ReactNode } from 'react';

import { render } from '@testing-library/react';

import { Provider } from '../3-custom/Provider/Provider';

export function renderWithProviders<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container
>(
  ui: ReactElement,
  renderOptions: RenderOptions<Q, Container, BaseElement> = {}
): RenderResult<Q, Container, BaseElement> {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return <Provider>{children}</Provider>;
  }
  return render<Q, Container, BaseElement>(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}
