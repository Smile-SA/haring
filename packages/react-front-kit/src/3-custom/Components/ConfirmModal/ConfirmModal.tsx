'use client';

import type { ModalProps } from '@mantine/core';
import type { ReactElement } from 'react';

import { Button, Modal } from '@mantine/core';

import { useStyles } from './ConfirmModal.style';

interface IConfirmModalProps extends ModalProps {
  cancelColor?: string;
  cancelLabel?: string;
  confirmColor?: string;
  confirmLabel?: string;
  onCancel?: () => void;
  onConfirm?: () => void;
  title?: string;
}

export function ConfirmModal(props: IConfirmModalProps): ReactElement {
  const {
    cancelColor = 'gray',
    cancelLabel = 'Cancel',
    confirmColor = 'red',
    confirmLabel = 'Confirm',
    onCancel,
    onConfirm,
    children,
    title,
    ...modalProps
  } = props;
  const { classes } = useStyles();
  return (
    <Modal
      centered
      classNames={{
        body: classes.modalBody,
        content: classes.modalContent,
        header: classes.modalHeader,
      }}
      radius={16}
      size="lg"
      {...modalProps}
    >
      <>
        <div className={classes.modalTitleContainer}>
          {title ? <h2>{title}</h2> : null}
          {children}
        </div>
        <div className={classes.modalButtonsContainer}>
          <Button
            className={classes.buttonLeftModal}
            classNames={{
              root: classes.buttonGrey,
            }}
            color={cancelColor}
            onClick={onCancel}
          >
            {cancelLabel}
          </Button>
          <Button
            classNames={{
              root: classes.buttonRemoveRoot,
            }}
            color={confirmColor}
            onClick={onConfirm}
          >
            {confirmLabel}
          </Button>
        </div>
      </>
    </Modal>
  );
}
