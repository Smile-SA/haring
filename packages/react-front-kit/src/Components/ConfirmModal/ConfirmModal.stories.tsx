import type { Meta, StoryObj } from '@storybook/react';

import { Button, DEFAULT_THEME } from '@mantine/core';
import { useStorybookArgsConnect } from '@smile/react-front-kit-shared/storybook-utils';
import { useArgs } from '@storybook/preview-api';

import { ConfirmModal as Cmp } from './ConfirmModal';

const colorOptions = Object.keys(DEFAULT_THEME.colors);

const meta = {
  argTypes: {
    cancelColor: {
      control: 'select',
      options: colorOptions,
    },
    confirmColor: {
      control: 'select',
      options: colorOptions,
    },
    opened: { control: 'boolean' },
  },
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onCancel: { opened: false },
        onClose: { opened: false },
        onConfirm: { opened: false },
      });
      const [, setArgs] = useArgs();
      return (
        <>
          <Button onClick={() => setArgs({ opened: true })}>Open modal</Button>
          <Story args={{ ...args }} />
        </>
      );
    },
  ],
  tags: ['autodocs'],
  title: '3-custom/Components/ConfirmModal',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const ConfirmModal: IStory = {
  args: {
    cancelColor: 'black',
    cancelLabel: 'Cancel',
    children: 'Are you sure you want to delete this item?',
    confirmColor: 'black',
    confirmLabel: 'Remove',
    opened: false,
    title: 'Remove ?',
  },
};
