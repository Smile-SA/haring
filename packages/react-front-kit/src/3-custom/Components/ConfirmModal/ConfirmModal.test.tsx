import { renderWithProviders } from '../../../utils/tests';

import { ConfirmModal } from './ConfirmModal';

describe('ConfirmModal', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <ConfirmModal
        cancelColor="gray"
        cancelLabel="Annuler"
        confirmColor="red"
        confirmLabel="Supprimer"
        onCancel={
          // eslint-disable-next-line no-console
          () => console.log('onCancel')
        }
        // eslint-disable-next-line no-console
        onClose={() => console.log('yolo')}
        onConfirm={
          // eslint-disable-next-line no-console
          () => console.log('onConfirm')
        }
        opened={false}
        title="Supprimer ?"
      >
        Voulez vous supprimer cette élément ?
      </ConfirmModal>,
    );
    expect(container).toMatchSnapshot();
  });
});
