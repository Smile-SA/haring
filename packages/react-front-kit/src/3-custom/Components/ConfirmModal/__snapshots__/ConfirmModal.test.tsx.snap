// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal matches snapshot 1`] = `
<div
  aria-hidden="true"
  data-focus-id="mantine-f4bipx4bi"
/>
`;
